### ISSUE TITLE
Accumulators inconsistent with PyTorch convention

### ISSUE THREAD

**Issue by f-dangel** [f-dangel/singd#18]
The PyTorch [convention](https://pytorch.org/docs/stable/notes/amp_examples.html#gradient-accumulation) is to divide the loss by the number of iterations (number of micro-batches) that are accumulated into a mini-batch gradient, and by the number of processes:
```python
loss /= iters_to_accumulate * num_procs
```

Instead, our [example](https://github.com/f-dangel/singd/blob/6f3ae8a916c57e46e5ee6dd98028bce2a1058855/examples/singd_and_adamw_full.py#L151) on how to use gradient accumulation does not do apply this division and is therefore inconsistent. To fix this, we have to rewrite the accumulator [`update`](https://github.com/f-dangel/singd/blob/6f3ae8a916c57e46e5ee6dd98028bce2a1058855/singd/optim/accumulator.py#L25-L42) function.

For the documentation, we should mention that it is crucial to use the PyTorch convention when accumulating over micro-batches.
